File: c0/p5/p5_ans.py
n, q = map(int, input().split())
shops = []
for i in range(n):
    a, s = input().split()
    shops.append([a, int(s)])
for i in range(q):
    x = int(input())
    ans = ["", 1e9]
    for shop in shops:
        if abs(shop[1] - x) < abs(ans[1] - x):
            ans = shop
    print(ans[0])

File: c0/p5/p5_ans.cpp
#include <iostream>
#define psi pair<string, int>
#define fi first
#define se second
using namespace std;

const int MAXSZ = 1005;

int n, q;
psi shops[MAXSZ];

signed main() {
    ios_base::sync_with_stdio(0);cin.tie(0);

    cin >> n >> q;
    for (int i = 0; i < n; i ++) {
        cin >> shops[i].fi >> shops[i].se;
    }

    for (int i = 0, x; i < q; i ++) {
        cin >> x;
        psi ans = {"", 1e9};
        for (int j = 0; j < n; j ++) {
            if (abs(shops[j].se - x) < abs(ans.se - x)) {
                ans = shops[j];
            }
        }
        cout << ans.fi << '\n';
    }
    return 0;
}